lib/spack/spack/test/concretize_requirements.py

```diff
@@ -124,6 +124,62 @@ def test_requirement_isnt_optional(concretize_scope, test_repo):
         Spec("x@1.1").concretize()
 
 
+def test_requirement_adds_new_version(
+    concretize_scope, test_repo, mock_git_version_info, monkeypatch
+):
+    if spack.config.get("config:concretizer") == "original":
+        pytest.skip("Original concretizer does not support configuration" " requirements")
+
+    repo_path, filename, commits = mock_git_version_info
+    monkeypatch.setattr(
+        spack.package_base.PackageBase, "git", "file://%s" % repo_path, raising=False
+    )
+
+    a_commit_hash = commits[0]
+    conf_str = """\
+packages:
+  v:
+    require: "@{0}=2.2"
+    version: ["2.2"]
```

Comment:
We should add documentation about this being necessary

Reply:
Added in https://github.com/spack/spack/pull/35057/commits/8886e3d6f2b01db494013c2abea7b082e84fc425